pkg/util/util.go

```diff
@@ -157,30 +157,38 @@ func ResourceRequirement(spec v1alpha1.ContainerSpec, defaultRequests ...corev1.
 		if rr.Requests == nil {
 			rr.Requests = make(map[corev1.ResourceName]resource.Quantity)
 		}
-		if q, err := resource.ParseQuantity(spec.Requests.CPU); err != nil {
-			glog.Errorf("failed to parse CPU resource %s to quantity: %v", spec.Requests.CPU, err)
-		} else {
-			rr.Requests[corev1.ResourceCPU] = q
+		if spec.Requests.CPU != nil {
```

Comment:
`spec.Requests.CPU`'s type is [`string`](https://github.com/weekface/tidb-operator/blob/fea4bd22345e55bf2967c63530ee831c8c32b5b7/pkg/apis/pingcap.com/v1alpha1/types.go#L156), can't compared with `nil`, replace `nil` with `""` instead